plugin/helpers_kv.go

```diff
@@ -129,3 +130,91 @@ func (p *HelpersImpl) KVSetWithExpiryJSON(key string, value interface{}, expireI
 
 	return nil
 }
+
+type kvListOptions struct {
+	KVListPerPage int
+	Prefix        string
+	Checker       func(key string) (keep bool, err error)
+}
+
+// KVListOption represents a single input option for KVListWithOptions
+type KVListOption func(*kvListOptions)
+
+// WithPrefix will filter out any key that doesn't start with the given string.
+func WithPrefix(prefix string) KVListOption {
+	return func(args *kvListOptions) {
+		args.Prefix = prefix
+	}
+}
+
+// WithChecker allows for a custom filter function to determine which keys to return.
+// Returning true will keep the key and false will filter it out.  Returning an error
+// will halt KVListWithOptions immediately and pass the error up (with no other results).
+func WithChecker(f func(key string) (keep bool, err error)) KVListOption {
+	return func(args *kvListOptions) {
+		args.Checker = f
+	}
+}
+
+// WithKVListPerPage is for unit testing KVListWithOptions.  This value defaults to 100 and shouldn't
+// be overriden outside of testing.
+func WithKVListPerPage(count int) KVListOption {
```

Comment:
Could you please elaborate why you need this? I'm hesitant to add a method to the public API just for testing.

What do you think about just testing the case where more the 100 elements are returned in total?

Reply:
I modeled the concept after the original PR that had this flexibility for testing.  This was the cleanest way I  could think to do it, but it still ends up being a bit dirty.   I'll move it to a constant and update the tests.